crates/core/src/operations/optimize.rs

```diff
@@ -74,6 +74,14 @@ pub struct Metrics {
     pub preserve_insertion_order: bool,
 }
 
+/// Information to be committed by the optimizer.
+pub struct CommitInfo {
```

Comment:
there is a native deltalake action also named commitInfo, so it would be good to name it differently here. perhaps CommitContext, CommitHandle or something like that.